lang/cs/Org.Apache.REEF.Client/YARN/RESTClient/JobResourceUploader.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.IO;
+using Org.Apache.REEF.Client.Common;
+using Org.Apache.REEF.Client.Yarn;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Client.YARN.RestClient
+{
+    public sealed class JobResourceUploader : IJobResourceUploader
+    {
+        private static readonly Logger Log = Logger.GetLogger(typeof(LegacyJobResourceUploader));
+        private static readonly DateTime Epoch = new DateTime(1970, 1, 1, 0, 0, 0, 0);
+        private readonly IJobSubmissionDirectoryProvider _jobSubmissionDirectoryProvider;
+        private readonly IResourceArchiveFileGenerator _resourceArchiveFileGenerator;
+        private readonly IFileSystem _fileSystem;
+
+        [Inject]
+        private JobResourceUploader(
+            IJobSubmissionDirectoryProvider jobSubmissionDirectoryProvider,
+            IResourceArchiveFileGenerator resourceArchiveFileGenerator,
+            IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem;
+            _resourceArchiveFileGenerator = resourceArchiveFileGenerator;
+            _jobSubmissionDirectoryProvider = jobSubmissionDirectoryProvider;
+        }
+
+        public JobResource UploadJobResource(string driverLocalFolderPath)
+        {
+            driverLocalFolderPath = driverLocalFolderPath.TrimEnd('\\') + @"\";
+            var driverUploadPath = _jobSubmissionDirectoryProvider.GetJobSubmissionDirectory().TrimEnd('/') + @"/";
```

Comment:
An orthogonal issue, but ``GetJobSubmissionDirectly()`` should probably specify whether it's local or remote in its naming. I'm always a bit confused.